examples/roundtrip/ping.cpp

```diff
@@ -250,18 +256,23 @@ int main (int argc, char *argv[])
       use_listener ? dds::core::status::StatusMask::data_available() : dds::core::status::StatusMask::none());
 
   dds::core::Duration waittime = dds::core::Duration::from_secs(static_cast<double>(timeOut ? timeOut : 5));
-  if (!match_readers_and_writers(reader, writer, waittime))
+  if (sendTerm) {
+    std::cout << "sending termination command to pong program\n" << std::flush;
+    writer.dispose_instance(roundtrip_msg);
     return EXIT_FAILURE;
```

Comment:
Likewise, why is sending a TERM sample reason to terminate the application with EXIT_FAILURE?